core/modules/qdisp/testQDisp.cc

```diff
@@ -0,0 +1,412 @@
+	// Test class Executive::add
+	// Modeled after ccontrol::UserQuery::submit()
+	ResourceUnit ru;
+	std::string chunkResultName = "mock";
+	boost::shared_ptr<rproc::InfileMerger> infileMerger;
+	boost::shared_ptr<ChunkMsgReceiverMock> cmr = ChunkMsgReceiverMock::newInstance(chunkId.get());
+	ccontrol::MergingRequester::Ptr mr = boost::make_shared<ccontrol::MergingRequester>(cmr, infileMerger, chunkResultName);
+	std::string msg = millisecs;
+	boost::chrono::system_clock::time_point start = boost::chrono::system_clock::now();
+	std::vector<qdisp::Executive::Spec> s(copies);
+	for (int j=0; j < copies; ++j) {
+		s[j].resource = ru;
+		s[j].request = msg;
+		s[j].requester = mr;
+		ex.add(sequence.incr(), s[j]);
+	}
+	ex.join();
+	boost::chrono::duration<double> secs = boost::chrono::system_clock::now() - start;
+	double ret = secs.count();
+	return ret;
+}
+
+class WorkerTest {
+public:
+	double _retVal;
+	void operator()(qdisp::Executive &ex, SequentialInt &sequence, SequentialInt &chunkId, std::string &millisecs, int copies) {
+		_retVal = executiveTest(ex, sequence, chunkId, millisecs, copies);
+	}
+};
+
+// This function is run in a separate thread to fail the test if it takes too long
+// for the jobs to complete.
+void timeoutFunc(qdisp::Executive &ex, int jobs, int millisecs) {
+	LOGF_INFO("timeoutFunc");
+	boost::this_thread::sleep(boost::posix_time::milliseconds(millisecs));
+	LOGF_INFO("timeoutFunc sleep over %1% empty=%2%" % millisecs % ex._empty.get());
+	BOOST_REQUIRE(ex._empty.get() == true);
+}
+
+BOOST_AUTO_TEST_CASE(Executive) {
+	LOGF_INFO("Executive test 1");
+	// Modeled after ccontrol::UserQuery::submit()
+	std::string str = qdisp::Executive::Config::getMockStr();
+	qdisp::Executive::Config::Ptr conf = boost::make_shared<qdisp::Executive::Config>(str);
+	boost::shared_ptr<qdisp::MessageStore> ms = boost::make_shared<qdisp::MessageStore>();
+	qdisp::Executive ex(conf, ms);
+	SequentialInt sequence(0);
+	SequentialInt chunkId(1234);
+	int jobs = 0;
+	// test single instance
+	int millisInt = 400;
+	std::string millis(boost::lexical_cast<std::string>(millisInt));
+	double duration = executiveTest(ex, sequence, chunkId, millis, 1);
+	++jobs;
+	LOGF_INFO("duration single %1%" % duration);
+	BOOST_CHECK(ex._empty.get() == true);
+	// test adding 4 jobs
+	duration = executiveTest(ex, sequence, chunkId, millis, 4);
+	jobs += 4;
+	LOGF_INFO("duration four %1%" % duration);
+	BOOST_CHECK(ex._empty.get() == true);
+	// test multiple threads
+	int threadCount = 10;
+	int jobsPerThread = 2;
+	std::vector<WorkerTest> workers(threadCount);
+	std::vector<boost::shared_ptr<boost::thread> > threads(threadCount);
+	// Run half with long sleep value.
+	int j = 0;
+	for(; j<threadCount/2; ++j) {
+		threads[j] = boost::make_shared<boost::thread>(&WorkerTest::operator(), &(workers[j]), boost::ref(ex), boost::ref(sequence),
```

Comment:
Hey, those are some nice cards there. But really, I'm just faster at reading code in narrow windows and I like having the side-by-side comparison (for meld or other diff tools) fit on my small laptop screen with a "reasonable" font size. I'd rather sit on the porch and say 'get off my lawn' than scroll right and lose context, okay? ;)